lib/rules/no-unused-services.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+
+const types = require('../utils/types');
+const emberUtils = require('../utils/ember');
+const propertyGetterUtils = require('../utils/property-getter');
+
+function getErrorMessage(serviceName) {
+  return `The service \`${serviceName}\` is not referenced in this JS file, but it may still be referenced in the corresponding HBS file. Please remove this service if it is not being used within the HBS file.`;
+}
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow unused service injections',
+      category: 'Miscellaneous',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-unused-services.md',
+    },
+    fixable: 'code',
+    schema: [],
+  },
+
+  ERROR_MESSAGE: getErrorMessage,
+
+  create(context) {
+    const unusedServicesMapping = {};
+
+    return {
+      // foo: service('bar')
+      Property(node) {
+        if (emberUtils.isInjectedServiceProp(node)) {
+          const name = node.key.name;
+          unusedServicesMapping[name] = node;
+        }
+      },
+      // @service('bar') foo;
+      ClassProperty(node) {
+        if (emberUtils.isInjectedServiceProp(node)) {
+          const name = node.key.name;
+          unusedServicesMapping[name] = node;
+        }
+      },
+      // this.foo...
+      MemberExpression(node) {
+        if (isEmpty(unusedServicesMapping) || !propertyGetterUtils.isSimpleThisExpression(node)) {
+          return;
+        }
+        const name = node.property.name;
+        delete unusedServicesMapping[name];
+      },
+      // 1. get(this, 'foo...');
+      // 2. getProperties(this, [..., 'foo..', ...]); or getProperties(this, ..., 'foo..', ...);
+      // 3. this.get('foo...');
+      // 4. this.getProperties([..., 'foo..', ...]); or this.getProperties(..., 'foo..', ...);
+      CallExpression(node) {
+        if (isEmpty(unusedServicesMapping)) {
+          return;
+        }
+
+        if (node.callee.name === 'get' && types.isThisExpression(node.arguments[0])) {
```

Comment:
Anytime we access a property like this, it's best practice to check the node type first. In this case, we need to check `node.callee.type === 'Identifier'` before accessing `node.callee.name`.